lib/puppet/provider/gluster_peer/gluster_peer.rb

```diff
@@ -0,0 +1,58 @@
+require File.expand_path(File.join(File.dirname(__FILE__), '..', 'gluster'))
+
+Puppet::Type.type(:gluster_peer).provide(
+  :gluster_peer, :parent => Puppet::Provider::Gluster) do
+
+  has_command(:gluster, 'gluster') do
+    environment :HOME => "/tmp"
+  end
+
+  def self.instances
+    all_peers.map do |peer|
+      new(
+        :name => peer,
+        :peer => peer,
+        :ensure => :present,
+      )
+    end
+  end
+
+  def self.prefetch(resources)
+    peers = instances
+    resources.keys.each do |name|
+      if provider = peers.find{ |peer| peer.name == name }
+        resources[name].provider = provider
+      end
+    end
+  end
+
+  def exists?
+    @property_hash[:ensure] == :present
+  end
+
+  def create
+    begin
+      gluster("peer", "probe", resource[:peer])
+    rescue Puppet::ExecutionFailure => e
+      # Prior to 3.7, gluster returned a less helpful error when it couldn't
+      # reach the peer it was probing.
+      ["Probe returned with Transport endpoint is not connected",
+       "Probe returned with unknown errno 107"].each do |msg|
```

Comment:
The only way to get that from the exception is to parse it out of the error message, and I haven't seen any evidence that it's useful for anything more than "something went wrong". I'd probably at least check it if it were easier to get at, though.